### PR TITLE
[DX] made deleteFileAfterSend() behavior explicit

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | **yes** |
| Deprecations? | no |
| Tests pass? | yes |
| License | MIT |
| Doc PR | No need, the docs already say [it's possible](http://symfony.com/doc/current/components/http_foundation.html#serving-files). |

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/BinaryFileResponse.php [hunk 1]
```diff
@@ -350,7 +350,7 @@ public static function trustXSendfileTypeHeader()
      *
      * @return BinaryFileResponse
      */
-    public function deleteFileAfterSend($shouldDelete)
+    public function deleteFileAfterSend($shouldDelete = true)
```
Comment (chalasr) [inline src/Symfony/Component/HttpFoundation/BinaryFileResponse.php]: This is a BC break forbidden by [our BC promise](http://symfony.com/doc/current/contributing/code/bc.html#changing-classes)

Comment (mickaelandrieu) [inline src/Symfony/Component/HttpFoundation/BinaryFileResponse.php]: Still, don't you think we should let this function remains "as it", even for Symfony 4 ?

Comment (chalasr) [inline src/Symfony/Component/HttpFoundation/BinaryFileResponse.php]: IMHO this method (named as is) should not take any argument and just enable the feature (giving the possibility to disable it with a different method), or be named `setDeleteFileAfterSend()`.

Btw regarding the doc, it's said this behavior can be enabled, not that is the default one

Comment (mickaelandrieu) [inline src/Symfony/Component/HttpFoundation/BinaryFileResponse.php]: @chalasr I'll improve the docs, and I'll redo this pull request (including a new `keepFileAfterSend()`) later.

Thanks for review :)

**Pull request closed without merging**